src/main/components/AssertionsForm.jsx

```diff
@@ -0,0 +1,120 @@
+import React, { useState, useContext } from 'react';
+import styled from 'styled-components';
+import { TestsContext } from '../testsContext';
+import Select from './InlineSelect.jsx';
+import InlineInput from './InlineInput.jsx';
+import Form from './InlineForm.jsx';
+import Button from './Button.jsx';
+
```

Comment:
Would redux be useful here? It seems like there is quite a bit of prop drilling and there looks like there is a lot of potential for actions and reducers to be used. For example all of the states further below can be set up with actions.

Reply:
Alas, they used context api in places as well. Now we need to decide whether to fully implement context, or overhaul for Redux. 

I'm leaning slightly toward more of a component overhaul, since it would, at the very least, squash all the bugs and console errors that already show up.